x/concentrated-liquidity/types/msgs.go

```diff
@@ -32,12 +32,22 @@ func (msg MsgCreatePosition) ValidateBasic() error {
 		return InvalidLowerUpperTickError{LowerTick: msg.LowerTick, UpperTick: msg.UpperTick}
 	}
 
-	if !msg.TokenDesired0.IsValid() || msg.TokenDesired0.IsZero() {
-		return fmt.Errorf("Invalid coins (%s)", msg.TokenDesired0.String())
+	if msg.TokensProvided.Empty() {
+		return fmt.Errorf("Empty coins provided (%s)", msg.TokensProvided.String())
 	}
 
-	if !msg.TokenDesired1.IsValid() || msg.TokenDesired1.IsZero() {
-		return fmt.Errorf("Invalid coins (%s)", msg.TokenDesired1.String())
+	if !msg.TokensProvided.IsValid() {
+		return fmt.Errorf("Invalid coins (%s)", msg.TokensProvided.String())
+	}
+
+	if len(msg.TokensProvided) > 2 {
+		return CoinLengthError{Length: len(msg.TokensProvided), MaxLength: 2}
+	}
```

Comment:
Since there are always 2 tokens expected, I think it is better UX to have 2 separate fields - one for token0 and another one for token1.

This way, people would not be able to shoot themselves in the foot by providing more tokens and failing the check here

Reply:
(this is what you were suggesting, I am just pointing out that just because the two fields make sense here does not mean we should also have it for createPosition as well)

Reply:
Ah it's because we cannot have a zero coin